feat(custom-resources):  AwsSdkCall can assume Role for cross-account custom resources

McDoit opened the pull request:
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*
#13915 

PR related to my feature request in #13915

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: b01dbd167028fc99a328bb7e340b1bf8107f7a78
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=89bab3af-cae4-4cc2-8e33-1257c746b13d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: c19ba0db2a1143d82042f792f1943cfc04d6aefe
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d3334260-187c-4567-b497-6921f8483dcb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 7f7b5ed909b191420e35bd76e201c667569906e2
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=17f95a97-a893-4ab4-9b4d-77dcd25d1c00%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: d55003c890e69a1c38bb71bfaf7e200659bf489f
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=32d0b08d-76eb-4e22-a20e-dad1768850f3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 819fd9f19ada8269f8da368e59fb8aeb350e46ce
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=03662ef1-f196-4980-b329-f14aa5947a95%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

McDoit commented:
I don't really know, but I guess it is expected that the hash get updated with this change right?

```
@aws-cdk/aws-route53: Verifying integ.vpc-endpoint-service-domain-name.js against integ.vpc-endpoint-service-domain-name.expected.json ... CHANGED.
@aws-cdk/aws-route53: Parameters
@aws-cdk/aws-route53: [-] Parameter AssetParameters0625b1566df06e0ffd948f0f65f97a3d22d48242e66196d3f72b480f5309b343S3Bucket65227904: {"Type":"String","Description":"S3 bucket for asset \"0625b1566df06e0ffd948f0f65f97a3d22d48242e66196d3f72b480f5309b343\""}
@aws-cdk/aws-route53: [-] Parameter AssetParameters0625b1566df06e0ffd948f0f65f97a3d22d48242e66196d3f72b480f5309b343S3VersionKey3AF0E7DF: {"Type":"String","Description":"S3 key for asset version \"0625b1566df06e0ffd948f0f65f97a3d22d48242e66196d3f72b480f5309b343\""}
@aws-cdk/aws-route53: [-] Parameter AssetParameters0625b1566df06e0ffd948f0f65f97a3d22d48242e66196d3f72b480f5309b343ArtifactHash0C561FF5: {"Type":"String","Description":"Artifact hash for asset \"0625b1566df06e0ffd948f0f65f97a3d22d48242e66196d3f72b480f5309b343\""}
@aws-cdk/aws-route53: [+] Parameter AssetParameters08eb0854a4c44f0c7a1713067d89113a3d4b8ba7aeac35d59bcb4a9e71edd33dS3BucketC698AA34: {"Type":"String","Description":"S3 bucket for asset \"08eb0854a4c44f0c7a1713067d89113a3d4b8ba7aeac35d59bcb4a9e71edd33d\""}
@aws-cdk/aws-route53: [+] Parameter AssetParameters08eb0854a4c44f0c7a1713067d89113a3d4b8ba7aeac35d59bcb4a9e71edd33dS3VersionKey2F8E8BC5: {"Type":"String","Description":"S3 key for asset version \"08eb0854a4c44f0c7a1713067d89113a3d4b8ba7aeac35d59bcb4a9e71edd33d\""}
@aws-cdk/aws-route53: [+] Parameter AssetParameters08eb0854a4c44f0c7a1713067d89113a3d4b8ba7aeac35d59bcb4a9e71edd33dArtifactHash768B7C58: {"Type":"String","Description":"Artifact hash for asset \"08eb0854a4c44f0c7a1713067d89113a3d4b8ba7aeac35d59bcb4a9e71edd33d\""}
@aws-cdk/aws-route53: Resources
@aws-cdk/aws-route53: [~] AWS::Lambda::Function AWS679f53fac002430cb0da5b7982bd22872D164C4C 
@aws-cdk/aws-route53:  â””â”€ [~] Code
@aws-cdk/aws-route53:      â”œâ”€ [~] .S3Bucket:
@aws-cdk/aws-route53:      â”‚   â””â”€ [~] .Ref:
@aws-cdk/aws-route53:      â”‚       â”œâ”€ [-] AssetParameters0625b1566df06e0ffd948f0f65f97a3d22d48242e66196d3f72b480f5309b343S3Bucket65227904
@aws-cdk/aws-route53:      â”‚       â””â”€ [+] AssetParameters08eb0854a4c44f0c7a1713067d89113a3d4b8ba7aeac35d59bcb4a9e71edd33dS3BucketC698AA34
@aws-cdk/aws-route53:      â””â”€ [~] .S3Key:
@aws-cdk/aws-route53:          â””â”€ [~] .Fn::Join:
@aws-cdk/aws-route53:              â””â”€ @@ -8,7 +8,7 @@
@aws-cdk/aws-route53:                 [ ]   "Fn::Split": [
@aws-cdk/aws-route53:                 [ ]     "||",
@aws-cdk/aws-route53:                 [ ]     {
@aws-cdk/aws-route53:                 [-]       "Ref": "AssetParameters0625b1566df06e0ffd948f0f65f97a3d22d48242e66196d3f72b480f5309b343S3VersionKey3AF0E7DF"
@aws-cdk/aws-route53:                 [+]       "Ref": "AssetParameters08eb0854a4c44f0c7a1713067d89113a3d4b8ba7aeac35d59bcb4a9e71edd33dS3VersionKey2F8E8BC5"
@aws-cdk/aws-route53:                 [ ]     }
@aws-cdk/aws-route53:                 [ ]   ]
@aws-cdk/aws-route53:                 [ ] }
@aws-cdk/aws-route53:                 @@ -21,7 +21,7 @@
@aws-cdk/aws-route53:                 [ ]   "Fn::Split": [
@aws-cdk/aws-route53:                 [ ]     "||",
@aws-cdk/aws-route53:                 [ ]     {
@aws-cdk/aws-route53:                 [-]       "Ref": "AssetParameters0625b1566df06e0ffd948f0f65f97a3d22d48242e66196d3f72b480f5309b343S3VersionKey3AF0E7DF"
@aws-cdk/aws-route53:                 [+]       "Ref": "AssetParameters08eb0854a4c44f0c7a1713067d89113a3d4b8ba7aeac35d59bcb4a9e71edd33dS3VersionKey2F8E8BC5"
@aws-cdk/aws-route53:                 [ ]     }
@aws-cdk/aws-route53:                 [ ]   ]
@aws-cdk/aws-route53:                 [ ] }
@aws-cdk/aws-route53: Error: Some stacks have changed. To verify that they still deploy successfully, run: 'yarn integ integ.vpc-endpoint-service-domain-name.js'
@aws-cdk/aws-route53:     at main (/codebuild/output/src066921871/src/github.com/aws/aws-cdk/tools/cdk-integ-tools/bin/cdk-integ-assert.js:36:15)
@aws-cdk/aws-route53: Error: cdk-integ-assert exited with error code 1
```

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: d14c8a7ecdfc6ae35dd4caa1abd24452c9177ec6
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=04c95122-a2aa-453b-bfe8-579838fcd2f0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 2b481d09fecde3971762c4a6ed7672e87912a688
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9db76048-e6e1-47b7-9492-a110a9e0ab4e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 347afdd38a975d13f3e9842fc086ee1733b97a34
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cf150644-b799-425e-8c36-77a7bd38cccb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 093bbb852726b9d56585c36e89d7ec282ef2e07f
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=84bea367-4aff-421f-a5b6-d768f0736bc7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: ccdafc22423526f10c69f5e5182b35253ab2ce06
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0e484545-f248-4b5d-86ec-db1e5f6675ec%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: b30cc69a93a49aed94c61fd0dfe4e564c60d4f07
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=273ebbaa-d003-4ce2-9bea-379cb99e33c7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: d490fb54c7ae9b89ee835d4f32421a75f424d672
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5666b1a1-c3ee-442a-bca0-45fc10a44f94%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 1d53413975dd71d4ac2b88805220869cbc9a9745
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b3ac08b5-f51d-45ef-9b50-83a623116401%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: bded8f293b8792aab438012e5d596f60eb957d75
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0dbab8a8-5d2a-4ab2-864a-c5b214ebc1df%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 70278d26a87b52b0bc58e0099fb406e2e2ea52ca
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2f2c3b7e-bf86-491a-a21d-5ed599ff68a1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

rix0rrr commented on packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts:
```diff
@@ -130,6 +130,17 @@ export interface AwsSdkCall {
+   *
+   * @default - run without assuming role
+   */
+  readonly assumedRoleArn?: string;
```
For consistency's sake this should be an `IRole`.

McDoit replied:
Will that be serialized correctly as a RoleArn when sent to the lambda, or do the lambda then receive a IRole to get the RoleArn from?

rix0rrr commented on packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts:
```diff
@@ -115,10 +115,42 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
+      if (call.assumedRoleArn) {
+
+        const stsService = new (AWS as any).STS({
+          apiVersion: call.apiVersion,
```
The API version doesn't go here.

rix0rrr replied:
But instead of this manual call, I'd prefer you use `ChainableTemporaryCredentials`.

https://docs.aws.amazon.com/ko_kr/AWSJavaScriptSDK/latest/AWS/ChainableTemporaryCredentials.html

See here for example usage:

https://github.com/aws/aws-cdk/blob/master/packages/aws-cdk/lib/api/aws-auth/sdk-provider.ts/#L314

McDoit replied:
Magic, I like it! Changed to using it now

rix0rrr commented on packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts:
```diff
@@ -115,10 +115,42 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
+          sessionToken: assumeRoleResponse.Credentials.SessionToken,
+        });
+
+        awsService = new AWS[call.service]({
```
Please dedupe between this and the other instantiation.

McDoit replied:
Is this still valid with using `ChainableTemporaryCredentials`, or what does dedupe mean in this case?

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 2c3a43a30e1801482a43f54ef62ae4767bc9b4c6
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=351d97c5-8181-4b63-a47d-0b27183a8411%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 60df770aa42c2422b020b75c6e3a4a9fb1d125d7
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4f37d350-b5d8-4c99-ab94-ad6121be953e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 4cd8baaacfcaf4c689e73e60372c0a8270b47a10
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cd4e553e-8447-4ac8-b9b0-932bae1f6251%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 89c063dc69f98f025fb6d689ea259b1729887f20
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2277ac5d-e2c0-46cc-abce-d745c9f4f295%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 14f2b33a74386356c6c3c0e96a3e32df60d35031
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=97f510cf-9ac9-469e-a9e0-e9b419602557%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 5e3f5b4cfe786783374ad713ac7a9c083ee08b65
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9d72dca2-8a0f-44e9-8261-f5570fede76d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: c8cb806e499cee18fd1abd77beb068cdb088ffe4
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=918d10ba-b117-4116-98a0-5222b1bd2ab8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: b9c0d2e91e6a22955bd7ec4a63f8ceac3659914b
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cdbe7915-4970-42a5-97c3-53366c66b6e5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

McDoit commented:
@rix0rrr The build after I resolved some conflicts ended in some very strange test fail error, on a component I didn't touch
Solved it with a fake commit that I reverted

```
aws-cdk: Summary of all failing tests
aws-cdk: FAIL test/version.test.js (6.818 s)
aws-cdk:   â— Return later version when exists & skip recent re-check
aws-cdk:     expect(received).not.toBeNull()
aws-cdk:     Received: null
aws-cdk:       48 |   const cache = new VersionCheckTTL(tmpfile(), 100);
aws-cdk:       49 |   const result = await latestVersionIfHigher('0.0.0', cache);
aws-cdk:     > 50 |   expect(result).not.toBeNull();
aws-cdk:          |                      ^
aws-cdk:       51 |   expect((result as string).length).toBeGreaterThan(0);
aws-cdk:       52 |
aws-cdk:       53 |   const result2 = await latestVersionIfHigher('0.0.0', cache);
aws-cdk:       at Object.<anonymous>.test (test/version.test.ts:50:22)
aws-cdk: Test Suites: 1 failed, 36 passed, 37 total
aws-cdk: Tests:       1 failed, 274 passed, 275 total
aws-cdk: Snapshots:   0 total
aws-cdk: Time:        12.454 s
aws-cdk: Ran all test suites.
aws-cdk: Error: /codebuild/output/src736182070/src/github.com/aws/aws-cdk/node_modules/jest/bin/jest.js exited with error code 1
aws-cdk: Tests failed. Total time (13.5s) | /codebuild/output/src736182070/src/github.com/aws/aws-cdk/node_modules/jest/bin/jest.js (13.4s)
```

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 913cbf7dc376b654b4836ffd6defd13bf7a7e7b4
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1e362dc9-cdc8-4c8d-ac51-5bc892640f83%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 518353a00a8c6946e635fad3c10faa2f5de56e4b
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=365b96d6-4f76-458b-b1cb-53e90cd0111c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

gmerton-amazon commented:
What should the service principal be in the trust policy in order for the SdkCall to assume the role?
Is it lambda, or something like sdkcall.amazonaws.com?  It'd be great to include an example of this in the documentation, if it's not already there.
Thanks!
